src/server/game/Scripting/ScriptMgr.h

```diff
@@ -1472,6 +1472,16 @@ class PlayerScript : public ScriptObject
      */
     virtual void OnQuestAbandon(Player* /*player*/, uint32 /*questId*/) { }
 
+    /**
+     * @brief This hook called before other CanFlyChecks are applied
+     *
+     * @param player Contains information about the Player
+     * @param mapId Contains information about the current map id
+     * @param zoneId Contains information about the current zone
+     * @param bySpell Contains information about the spell that invoked the check
+     */
+    [[nodiscard]] virtual bool OnCanPlayerFlyInZone(Player const* /*player*/, uint32 /*mapId*/, uint32 /*zoneId*/, SpellInfo const* /*bySpell*/) { return true; }
```

Comment:
```suggestion
    [[nodiscard]] virtual bool OnCanPlayerFlyInZone(Player* /*player*/, uint32 /*mapId*/, uint32 /*zoneId*/, SpellInfo const* /*bySpell*/) { return true; }
```

Reply:
Do not need to use a constant pointer

Reply:
I have nothing against it, but what is the benefit of a non const pointer over a const pointer? In which usecase would you anticipate that the additional flexibility becomes useful?

Reply:
For scripts and other better `Player*`. All functions can be used

Reply:
Ah I see, sounds good. Well at least in theory. These changes do not successfully compile (at least not without further changes)
The reason is that the function is called like this:
 
```
`bool Player::canFlyInZone(uint32 mapid, uint32 zone, SpellInfo const* bySpell) const ` like: if (!sScriptMgr->OnCanPlayerFlyInZone(this, mapid,zone,bySpell))
    {
        return false;
    }
    
```
the object **this** already references a const * player. Which in turn stems from the fact, its called from within:
```
SpellCastResult SpellInfo::CheckLocation(uint32 map_id, uint32 zone_id, uint32 area_id, Player const* player /*= nullptr*/, bool strict /*= true*/) const
{....}
```
Essentially, if we want to provide this hook without the const restriction we need to loosen that restriction earlier in the call stack as well. That is not too hard to do, but the question is whether or not we really want it.

Reply:
I adjusted the code accordingly, now we don't have const * player anymore